core/src/main/java/org/elasticsearch/action/fieldstats/FieldStats.java

```diff
@@ -449,7 +450,11 @@ public String getMaxValue() {
 
         @Override
         protected java.lang.Long valueOf(String value) {
-            return dateFormatter.parser().parseMillis(value);
+            try {
+                return dateFormatter.parser().parseMillis(value);
+            } catch (Exception e) {
```

Comment:
@bleskes This is in the case if someone has configured a custom date field format. (see #14804) If that happens Kibana doesn't know about this and it is very hard for them to convert to mills since epoch. So adding a format option would help here a lot.